### ISSUE TITLE
How do I print the key-value pairs of a dictionary in python

### ISSUE THREAD

**Issue by AbhignyaTayi** [arungudelli/python.shiksha#112]
Python provides a built-in dictionary data structure, which allows us to store data in key-value pairs. Retrieving both the key and the value of a dictionary in Python is often necessary, especially when working with large data sets. In this article, we will discuss different ways of printing key-value pairs of a dictionary in Python with code and output.

## Printing key-value pairs of a dictionary in Python:

Python's dictionaries are data structures that store data in a key-value format. Sometimes, you may want to print the key-value pairs of a dictionary. There are many ways to do this in Python, and we'll cover five of the most common methods.

###  Method 1:  Using a `for` loop

One way to print the key-value pairs of a dictionary is to use a for loop to iterate over the keys. For each key, you can use the dictionary's get() method to retrieve the corresponding value. Here's an example:

Code : 

```
my_dict = {'apple': 2, 'banana': 3, 'orange': 4}

for key in my_dict:
    print(key, my_dict.get(key))

```
Output : 

```
apple 2
banana 3
orange 4


```
###  Method 2 : Using `items() `method

Python's dictionary object provides the `items()` method, which returns an iterable view of key-value pairs. You can use this method to print the key-value pairs of a dictionary.

Code : 

```
my_dict = {'apple': 2, 'banana': 3, 'orange': 4}

for key, value in my_dict.items():
    print(key, value)

```

Output : 

```
apple 2
banana 3
orange 4

```

###  Method 3 : Using `iteritems()` method

In Python 2, you can use the 'iteritems()' method, which returns an iterator over the dictionary's key-value pairs. This can be more efficient than using the `items()` method in some cases.

Code : 

```
my_dict = {'apple': 2, 'banana': 3, 'orange': 4}

for key, value in my_dict.iteritems():
    print(key, value)


```
Output : 

```
apple 2
banana 3
orange 4

```

###  Method 4: Using a list comprehension

You can also use list comprehension to print the key-value pairs of a dictionary.

Code : 

```
my_dict = {'apple': 2, 'banana': 3, 'orange': 4}

print([(key, value) for key, value in my_dict.items()])

```

Output : 

```
[('apple', 2), ('banana', 3), ('orange', 4)]

```

###  Method 5: Using a generator expression

Finally, you can use a generator expression to print the key-value pairs of a dictionary.

Code : 
```

my_dict = {'apple': 2, 'banana': 3, 'orange': 4}

print((key, value) for key, value in my_dict.items())

```

Output : 

```
<generator object <genexpr> at 0x7f1fb3c7f8e0>

```

## Conclusion 

In conclusion, printing the key-value pairs of a dictionary is an essential operation in Python programming. In this article, we have explored five different methods to achieve this, including using a for loop, the `items()` method, the `keys()` and `values()` methods, list comprehension, and using the`pprint` module.

All these methods have their unique advantages, and the choice of which one to use depends on the particular use case. As a Python programmer, having a good understanding of the various methods available for printing dictionary key-value pairs is a valuable skill that can help make your code more efficient and easier to read.
